enso-org/enso
# Allow downloading projects on GUI dev server

### Pull Request Description
- Fixes #9482
  - Add URL rewriting logic to `npm run dev` in `app/gui2/` so that "download" actions work

### Important Notes
- :warning: `config.yaml` HAS BEEN REMOVED in favor of `config.json`.
  - Most of the keys have been removed as they were only used by GUI1.
  - This is a REQUIRED change, because the vite dev server does not seem to like importing yaml files when running `vite.config.ts`.

### Testing instructions
- Just test downloads on every desktop app variant:
  - `npm run dev` on `app/gui2/`
  - `npm run watch:linux` (or `npm run watch:macos` or `npm run watch:windows` as appropriate) in `app/ide-desktop/lib/client/`
  - `./run ide build --skip-version-check --backend-source release --backend-release nightly` in repo root, and then running the resulting executable in `dist/ide/`
  - Download variants to test:
    - Download project by selecting and clicking the "download" icon in the drive bar at the top
    - Context menu button
    - Keyboard shortcut (see context menu of project asset)

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
